pkg/client/automation/client.go

```diff
@@ -109,36 +111,47 @@ func WithClientRequestLimiter(limiter *concurrency.Limiter) func(client *Client)
 }
 
 // List returns all automation objects
-func (a Client) List(resourceType ResourceType) (result *ListResponse, err error) {
+func (a Client) List(resourceType ResourceType) (result *[]Response, err error) {
 	a.limiter.ExecuteBlocking(func() {
 		result, err = a.list(resourceType)
 	})
 	return
 }
 
-func (a Client) list(resourceType ResourceType) (*ListResponse, error) {
-	// try to get the list of resources
-	resp, err := rest.Get(a.client, a.url+a.resources[resourceType].Path)
-	if err != nil {
-		return nil, fmt.Errorf("unable to list automation resources: %w", err)
-	}
+func (a Client) list(resourceType ResourceType) (*[]Response, error) {
+	var retVal []Response
+	var result ListResponse
+	result.Count = 1
 
-	// handle http error
-	if !resp.IsSuccess() {
-		return nil, ResponseErr{
-			StatusCode: resp.StatusCode,
-			Message:    "Failed to list automation objects",
-			Data:       resp.Body,
+	for len(retVal) < result.Count {
+		// try to get the list of resources
+		resp, err := get(a, resourceType, len(retVal))
+		if err != nil {
+			return nil, fmt.Errorf("unable to list automation resources: %w", err)
+		}
+
+		// handle http error
+		if !resp.IsSuccess() {
+			return nil, ResponseErr{
+				StatusCode: resp.StatusCode,
+				Message:    "Failed to list automation objects",
+				Data:       resp.Body,
+			}
 		}
+
+		// unmarshal and return result
+
+		err = json.Unmarshal(resp.Body, &result)
+		if err != nil {
+			return nil, err
+		}
+		retVal = append(retVal, result.Results...)
 	}
 
-	// unmarshal and return result
-	var result ListResponse
-	err = json.Unmarshal(resp.Body, &result)
-	if err != nil {
-		return nil, err
+	if len(retVal) != result.Count {
```

Comment:
We could technically get a case in which we have more return values than the last API result expects - if one was deleted between calls and the last `Count` ends up smaller than `retVal` + `results.Results`.

I think we should treat that case differently than getting fewer than expected. Possibly with just a warning log and returning the values normally. Wdyt?

Reply:
Good observation.
changed to warn